Add a --system argument for system-wide installation

PLPeeters opened the pull request:
This PR adds a `--system` argument for system-wide installation.

Since there doesn't seem to be a simple way to figure out the `$(prefix)` variable of the Git installation, I had to resort to a kind of hacky way to figure that one out in order to put the system-wide `gitattributes` file in the correct place (see the `_get_system_gitconfig_folder()` function).

I also fixed the path of the gitattributes file in the documentation when using the `--global` argument, which was incorrectly documented as being `.git/info/attributes` instead of `~/.config/git/attributes`.

I also took the liberty of adding a few newlines to improve readability.

PLPeeters commented:
I also just went ahead and added a fix and a test for the `--status` command which was apparently broken, along with some unification in the treatment of shell output (setting `text=True` everywhere and removing calls to `.decode()` that are no longer necessary with this). I can do a separate PR for that if you prefer, but since it intersects with my changes it was just easier for me to include it in here.

kynan reviewed (changes requested):
Thanks for your contribution! That's a useful feature.

Thanks also for the other fixes  / cleanup and for adding a test.

Just a minor nit, otherwise I think this is good to go!

PLPeeters commented on nbstripout/_nbstripout.py:
```diff
@@ -133,29 +142,65 @@ def write(nb, f):
+        output_lines = git_config_output.split('\n')
+
+        system_gitconfig_file_path = re.sub(r'^file:', '', output_lines[0].split('\t')[0])
+    except CalledProcessError as e:
```
Do you mean the cases where we would enter the `except` block? If so, the call fails if the config file does not exist, but when it does it prints the path where it expected to find it, which is why I parse the output to get the path.

kynan reviewed (approved):
Thanks, I think this is good to go now!

kynan reviewed (changes requested):
Sorry, spoke too soon, [this is not passing tests](https://travis-ci.org/github/kynan/nbstripout/builds/767845555):
1. looks like the `text` parameter to `check_output` is 3.7+ only. Can you use `universal_newlines` instead?
2. we're getting [lots of stray output](https://travis-ci.org/github/kynan/nbstripout/jobs/767845558): `fatal: --local can only be used inside a git repository`

PLPeeters commented:
@kynan Changed the text parameter as requested.

The stray output was due to the `git config` call to fetch the extra keys. I changed it to a simple `git config` when `nbstripout` is called without the `--system` or `--global` parameters, since that seemed to make the most sense.

I also fixed the Git test since it didn't run properly on my system even though it looks fine on Travis. Apparently the version of Git on macOS seems to insist on having a newline at the end of the attributes file (it threw an `*.txt is not a valid attribute name: .git/info/attributes:1` error on all the Git calls in that test for me).

kynan reviewed (approved):
Thanks, excellent work, really appreciate your contribution :)
